### ISSUE TITLE
🐛 BUG: TypeScript keeps loading intelligence status indefinitely 

### ISSUE THREAD

**Issue by ahmad-ali14** [withastro/language-tools#547]
### Describe the Bug

The typescript extension keeps loading intelligence status indefinitely with very high usage of CPU. 

Reloading the VSCode window seems to solve it for a while, then it will crash again.

I see this repeats when switching from astro to ts files or vice versa. 

I removed all typescript extensions and I'm using the following versions:

- Astro 2.5.5
- TypeScript 5.0.4
- Astro VsCode Extention 2.0.8

I see this error in `tsserver.log`:

```
Err 5091  [10:15:27.273] exit code:, 2
Info 5092 [10:15:27.273] -Astro Plugin- Error loading Astro file: /...path/[id]/RecommendedQueries/index.astro
Info 5093 [10:15:27.273] -Astro Plugin- Astro snapshot was not found after trying to load script snapshot for /...path/[id]/RecommendedQueries/index.astro
Info 5094 [10:15:27.273] FileWatcher:: Added:: WatchInfo: /...path/forms/GalleryForm.astro 500 undefined WatchType: Closed Script info
Info 5095 [10:15:27.280] -Astro Plugin- Successfully read Astro file contents of /...path/GalleryForm.astro
Info 5096 [10:15:27.286] -Astro Plugin- Successfully read Astro file contents of /...path/GalleryForm.astro
Info 5097 [10:15:27.339] -Astro Plugin- Successfully read Astro file contents of /...path/GalleryForm.astro
Info 5098 [10:15:28.215] -Astro Plugin- Successfully read Astro file contents of /...path/GalleryForm.astro
Info 5099 [10:15:33.514] -Astro Plugin- Successfully read Astro file contents of /...path/GalleryForm.astro
Info 5100 [10:15:33.514] -Astro Plugin- AstroSnapshot: /...path/GalleryForm.astro - patched scriptInfo
Info 5101 [10:15:33.514] FileWatcher:: Added:: WatchInfo: /...path/PropertyForm.astro 500 undefined WatchType: Closed Script info
Info 5102 [10:15:33.520] -Astro Plugin- Successfully read Astro file contents of /...path/PropertyForm.astro
Info 5103 [10:15:33.544] -Astro Plugin- Successfully read Astro file contents of /...path/PropertyForm.astro
Info 5104 [10:15:33.754] -Astro Plugin- Successfully read Astro file contents of /...path/PropertyForm.astro
Info 5105 [10:15:35.766] -Astro Plugin- Successfully read Astro file contents of /...path/PropertyForm.astro
```

The same error is repeated for a dozen of files in the project, that are not related to each other nor opened. 

The crash does not happen at all times, but most of the time. 

### Steps to Reproduce

1. open a fresh astro project
2. create ts/Astro files
3. move between Astro and Ts files and monitor CPU usage and tsserver logs.

**Comment by Princesseuh** [comment]
Does the same thing happen when using the normal release of the extension?

**Comment by Princesseuh** [comment]
Interesting, I'll take a look. Very sorry for the inconvenience!

**Comment by ahmad-ali14** [comment]
@Princesseuh Worth noting that I switched back to version 1.0.0 of the Astro VsCode extension (while keeping all other tools updated) and the problem seems to be resolved.

**Comment by johnsoncodehk** [comment]
Can you try if the version in this comment still has the problem? https://github.com/withastro/language-tools/issues/593#issuecomment-1612474903

**Comment by Princesseuh** [comment]
This should now be fixed in `2.1.0` of the extension. Please open another issue if this still happens!
